src/spatial/collision.js

```diff
@@ -483,46 +487,46 @@ Crafty.c("Collision", {
      *
      * @see Crafty.map#Crafty.map.search
      */
-    hit: function (component) {
-        var area = this._cbr || this._mbr || this,
-            results = Crafty.map.unfilteredSearch(area),
-            i = 0,
-            l = results.length,
-            dupes = {},
-            id, obj, oarea, key,
-            overlap = Crafty.rectManager.overlap,
-            hasMap = ('map' in this && 'containsPoint' in this.map),
-            finalresult = [];
-
+    _collisionHitDupes: [],
+    _collisionHitResults: [],
+    hit: function (component, finalresult) {
+        var area = this._cbr || this._mbr || this;
+        var results = this._collisionHitResults;
+        results.length = 0;
+        results = Crafty.map.unfilteredSearch(area, results);
+        var l = results.length;
         if (!l) {
             return null;
         }
+        var  i = 0,
+            dupes = this._collisionHitDupes,
+            id, obj;
+
+        finalresult = finalresult || [];
+        dupes.length = 0;
 
         for (; i < l; ++i) {
             obj = results[i];
-            oarea = obj._cbr || obj._mbr || obj; //use the mbr
 
             if (!obj) continue;
             id = obj[0];
 
             //check if not added to hash and that actually intersects
-            if (!dupes[id] && this[0] !== id && obj.__c[component] && overlap(oarea, area))
+            if (!dupes[id] && this[0] !== id && obj.__c[component]){
                 dupes[id] = obj;
-        }
-
-        for (key in dupes) {
-            obj = dupes[key];
-
-            if (hasMap && 'map' in obj) {
-                var SAT = this._SAT(this.map, obj.map);
-                SAT.obj = obj;
-                SAT.type = "SAT";
-                if (SAT) finalresult.push(SAT);
```

Comment:
Fun fact: when `_SAT` returned false, this code was previously trying to set the `obj` and `type` properties on that value.  This 'works' in javascript in the sense that it doesn't produce an error; instead, the boolean primitive is implicitly coerced to an object, the value is set on that object, and then that is ignored (since it isn't assigned or referneced anywhere.)  I think the intermediate object was probably getting optimized away, but this was at best very confusing.  (Here is [an article that describes what happens when you set properties on primitives.](https://javascriptweblog.wordpress.com/2010/09/27/the-secret-life-of-javascript-primitives/))